helm/sealed-secrets/templates/deployment.yaml

```
@@ -40,6 +40,9 @@ spec:
       securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
       {{- end }}
       serviceAccountName: {{ include "sealed-secrets.serviceAccountName" . }}
+      {{- if hasKey .Values "automountServiceAccountToken" }}
```

I would change the condition to use the same approach across the whole chart:

```suggestion
      {{- if .Values.automountServiceAccountToken }}
```

The same applies to the ServiceAccount file.

---

Done